parachains/runtimes/assets/statemine/src/weights/xcm/mod.rs

```diff
@@ -0,0 +1,180 @@
+			Self::Wild(_) => (MAX_ASSETS as Weight).saturating_mul(weight),
+		}
+	}
+}
+
+impl WeighMultiAssets for MultiAssets {
+	fn weigh_multi_assets(&self, weight: Weight) -> Weight {
+		(self.inner().into_iter().count() as Weight).saturating_mul(weight)
+	}
+}
+
+pub struct StatemineXcmWeight<Call>(core::marker::PhantomData<Call>);
+impl<Call> XcmWeightInfo<Call> for StatemineXcmWeight<Call> {
+	fn withdraw_asset(assets: &MultiAssets) -> Weight {
+		assets.weigh_multi_assets(XcmFungibleWeight::<Runtime>::withdraw_asset())
+	}
+	// Currently there is no trusted reserve
+	fn reserve_asset_deposited(_assets: &MultiAssets) -> Weight {
+		unimplemented!()
+	}
+	fn receive_teleported_asset(assets: &MultiAssets) -> Weight {
+		assets.weigh_multi_assets(XcmFungibleWeight::<Runtime>::receive_teleported_asset())
+	}
+	fn query_response(_query_id: &u64, _response: &Response, _max_weight: &u64) -> Weight {
+		XcmGeneric::<Runtime>::query_response()
+	}
+	fn transfer_asset(assets: &MultiAssets, _dest: &MultiLocation) -> Weight {
+		assets.weigh_multi_assets(XcmFungibleWeight::<Runtime>::transfer_asset())
+	}
+	fn transfer_reserve_asset(
+		assets: &MultiAssets,
+		_dest: &MultiLocation,
+		_xcm: &Xcm<()>,
+	) -> Weight {
+		assets.weigh_multi_assets(XcmFungibleWeight::<Runtime>::transfer_reserve_asset())
+	}
+	fn transact(
+		_origin_type: &OriginKind,
+		_require_weight_at_most: &u64,
+		_call: &DoubleEncoded<Call>,
+	) -> Weight {
+		XcmGeneric::<Runtime>::transact()
+	}
+	fn hrmp_new_channel_open_request(
+		_sender: &u32,
+		_max_message_size: &u32,
+		_max_capacity: &u32,
+	) -> Weight {
+		// XCM Executor does not currently support HRMP channel operations
+		Weight::MAX
+	}
+	fn hrmp_channel_accepted(_recipient: &u32) -> Weight {
+		// XCM Executor does not currently support HRMP channel operations
+		Weight::MAX
+	}
+	fn hrmp_channel_closing(_initiator: &u32, _sender: &u32, _recipient: &u32) -> Weight {
+		// XCM Executor does not currently support HRMP channel operations
+		Weight::MAX
+	}
+	fn clear_origin() -> Weight {
+		XcmGeneric::<Runtime>::clear_origin()
+	}
+	fn descend_origin(_who: &InteriorMultiLocation) -> Weight {
+		XcmGeneric::<Runtime>::descend_origin()
+	}
+	fn report_error(
+		_query_id: &QueryId,
+		_dest: &MultiLocation,
+		_max_response_weight: &u64,
+	) -> Weight {
+		XcmGeneric::<Runtime>::report_error()
+	}
+
+	fn deposit_asset(
+		assets: &MultiAssetFilter,
+		_max_assets: &u32, // TODO use max assets?
+		_dest: &MultiLocation,
+	) -> Weight {
+		assets.weigh_multi_assets(XcmFungibleWeight::<Runtime>::deposit_asset())
+	}
+	fn deposit_reserve_asset(
+		assets: &MultiAssetFilter,
+		_max_assets: &u32, // TODO use max assets?
+		_dest: &MultiLocation,
+		_xcm: &Xcm<()>,
+	) -> Weight {
+		assets.weigh_multi_assets(XcmFungibleWeight::<Runtime>::deposit_reserve_asset())
+	}
+	fn exchange_asset(_give: &MultiAssetFilter, _receive: &MultiAssets) -> Weight {
+		Weight::MAX // todo fix
```

Comment:
?

Reply:
resolved?

Reply:
yep, this is resolved, this stays as it was as it's not defined, so will have `weight::MAX`. @joepetrowski does this seem reasonable?

Reply:
As long as it's unimplemented it seems OK
https://github.com/paritytech/polkadot/blob/release-v0.9.28/xcm/xcm-executor/src/lib.rs#L504